packages/react/src/hooks/useStorage.tsx

```diff
@@ -0,0 +1,19 @@
+import { S3ProviderGetConfig, Storage } from '@aws-amplify/storage';
+import { useEffect, useState } from 'react';
+
+export const useStorageURL = (key: string, options: S3ProviderGetConfig) => {
+  const [url, setURL] = useState<string>();
+  const [error, setError] = useState<Error>();
+
+  const fetch = () => {
+    Storage.get(key, options).then(setURL).catch(setError);
+  };
+
+  useEffect(fetch, []);
+
+  return {
+    url,
```

Comment:
In your screenshot, `<Image src={url}>` was immediately used. But if this is an async call, what happens to images? Is there a "flash", or do we expect `<img>` components to have an empty space from `<img src="">`?

Reply:
At first render, `url` will be undefined. Users can read the value of `url` to determine what to render in that case.

```
const { url } = useStorageURL(key, options);

if (!url) {
  return <Placeholder />;
}

return <Image src={url} />
```

Reply:
That works for me! In documentation, the async nature of it should be clear so long as we do `const { isLoading, url } = useStorageUrl(key)` 👍